### PR TITLE
feat: reusable ipns verify

### PR DESCRIPTION
**Pull request by laurentsenta**
Sister PR in boxo: https://github.com/ipfs/boxo/pull/294

Extracts the ipns verification code out of kubo to boxo. This makes the verification code reusable, for example, with gateway-conformance testing (https://github.com/ipfs/gateway-conformance/pull/37)

Relates to https://github.com/ipfs/specs/issues/376

- [x] ⚠️ update with the correct boxo commit when the sister PR is merged

### CODE REVIEW

**Comment by laurentsenta** [comment]
(moving to draft until I fix the errors)

**Comment by laurentsenta** [comment]
Thanks for updating the PR @hacdias, that looks great!

**Review by Jorropo — APPROVED** [review]
LGTM, good to merge once merged and commit ID updated in go.mod here too.

**Review by hacdias — APPROVED** [review]

**Pull request merged**
